component-tools/src/main/java/org/talend/sdk/component/tools/DocBaseGenerator.java

```diff
@@ -244,8 +256,30 @@ private String doResolveProperty(final String propPath, final String paramRef) {
     }
 
     @Data
-    @AllArgsConstructor(access = PRIVATE)
-    protected static class Param {
+    protected static class UIInfo {
+
+        private final String section;
+
+        private final Set<String> nestedLayout = new HashSet<>();
+
+        public UIInfo(final String section, final String layout) {
+            this.section = section;
+            nestedLayout.add(layout);
+        }
+
+        /**
+         * We need a set of layouts since a complex type can have sub-options in differents layouts
```

Comment:
Not sure this comment is at the right position + javadoc is incomplete, can you complete it or drop it please ? (It is ok to remove it for me)